django-backend/fecfiler/scha_transactions/test_models.py

```diff
@@ -35,6 +35,31 @@ def test_get_scha_transaction_in_f3x_summary(self):
         )
         return trans
 
+    def test_parent_related_values(self):
+        t1, t2 = SchATransaction.objects.all()[:2]
+        t1.parent_transaction = t2
+        t1.update_parent_related_values(t2)
+        self.assertEqual(t1.back_reference_tran_id_number, t2.transaction_id)
+        self.assertEqual(t1.back_reference_sched_name, t2.form_type)
+
+    def test_generate_uid(self):
+        committee = CommitteeAccount()
+        f3x = F3XSummary(
+            committee_account=committee
+        )
+        trans = SchATransaction(
+            committee_account=committee,
+            report_id=f3x
+        )
+        self.assertFalse(trans.transaction_id)
+        trans.generate_unique_transaction_id()
+        self.assertTrue(trans.transaction_id)
+
+    def test_catches_uid_conflict(self):
+        trans = SchATransaction.objects.all()[0]
+        existing_uid = trans.transaction_id
+        self.assertTrue(SchATransaction.check_for_uid_conflicts(existing_uid))
+
     def test_save_and_delete(self):
```

Comment:
can you add an assertion that the `transaction_id` is defined after save?

Reply:
That's well worth adding.  Done.

Reply:
Well, once I swapped it from testing `id` to testing `transaction_id`...

Reply:
The fix is on different lines than the preview that Github is showing, so I'll leave it unresolved till you've reviewed it